cli/src/pcluster/api/controllers/image_logs_controller.py

```diff
@@ -0,0 +1,156 @@
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=W0613
+import dateutil
+
+from pcluster.api.controllers.common import configure_aws_region, convert_errors
+from pcluster.api.errors import BadRequestException
+from pcluster.api.models import (
+    GetImageLogEventsResponseContent,
+    GetImageStackEventsResponseContent,
+    ListImageLogStreamsResponseContent,
+    LogEvent,
+    LogStream,
+    StackEvent,
+)
+from pcluster.models.imagebuilder import ImageBuilder
+from pcluster.utils import to_iso_time
+
+
+def _validate_timestamp(val, ts_name):
+    try:
+        dateutil.parser.parse(val)
+    except Exception:
+        raise BadRequestException(
+            f"{ts_name} filter must be in the ISO 8601 format: YYYY-MM-DDThh:mm:ssTZD. "
+            "(e.g. 1984-09-15T19:20:30+01:00 or 1984-09-15)."
+        )
+
+
+@configure_aws_region()
+@convert_errors()
+def get_image_log_events(
+    image_id,
+    log_stream_name,
+    region=None,
+    next_token=None,
+    start_from_head=None,
+    limit=None,
+    start_time=None,
+    end_time=None,
+):
+    """
+    Retrieve the events associated with an image build.
+
+    :param image_id: Id of the image.
+    :type image_id: str
+    :param log_stream_name: Name of the log stream.
+    :type log_stream_name: str
+    :param region: AWS Region that the operation corresponds to.
+    :type region: str
+    :param next_token: Token to use for paginated requests.
+    :type next_token: str
+    :param start_from_head: If the value is true, the earliest log events are returned first. If the value is false,
+                            the latest log events are returned first. (Defaults to &#39;false&#39;.)
+    :type start_from_head: bool
+    :param limit: The maximum number of log events returned. If you don&#39;t specify a value, the maximum is as many
+                  log events as can fit in a response size of 1 MB, up to 10,000 log events.
+    :type limit:
+    :param start_time: The start of the time range, expressed in ISO8601 format
+                       (e.g. &#39;2021-01-01T20:00:00.000Z&#39;). Events with a timestamp equal to this time or later
+                       than this time are included.
+    :type start_time: str
+    :param end_time: The end of the time range, expressed in ISO8601 format (e.g. &#39;2021-01-01T20:00:00.000Z&#39;).
+                     Events with a timestamp equal to or later than this time are not included.
+    :type end_time: str
+
+    :rtype: GetImageLogEventsResponseContent
+    """
+    if start_time:
+        _validate_timestamp(start_time, "start_time")
+    if end_time:
+        _validate_timestamp(end_time, "end_time")
+
+    imagebuilder = ImageBuilder(image_id=image_id)
+    log_events = imagebuilder.get_log_events(
```

Comment:
what if image does not exist? I'm afraid in the model we are always throwing a generic ClusterActionError which gets translated into an InternalServiceException for the API. I believe we need to use more appropriate exceptions to have the correct status code from the API

Reply:
See comment on PR